sites/wonderland/blog/2023-08-12-a-zero-day-exploit-in-geb.md

```
@@ -0,0 +1,295 @@
+Once we resumed communication and became aware of the seriousness of the situation, we researched two possible solutions. The first, unfortunately, led us to an impasse. However, the second option was the most appropriate despite the technical challenges involved in the present ungovernance. This technical correction prevented us from being the cause of the closure of one of our ([and Vb!](https://x.com/kyoronut/status/1616955373851926533?s=20)) favorite stablecoins.
+
+The solution required an action by the DAO to authorize a contract and modify one of the few parameters that could still be modified. Then another question arose, **"How do we propose and vote on this action without publicly disclosing the bug?"**. The transactions necessary to execute the fix were to be discreetly sent along with the November FLX reward distribution, hoping no one would notice them.
+
+A week later, the [innocent proposal](https://www.tally.xyz/gov/reflexer-ungovernor/proposal/39) was approved, and the fix went into effect. With RAI (the most affected protocol and with the highest TVL) out of danger, we disclosed the bug to the other team affected by this design flaw: TAI. As TAI had fewer restrictions in terms of governance, the solution was more straightforward, and the next day, [another innocent proposal](https://www.tally.xyz/gov/tai/proposal/25715647870826724893384566403969478394242704320930112822710906676227083258387?chart=1) was submitted to Tally to modify the necessary parameters and make the bug unreachable in their protocol.
+
+This is the story of how Wonderland endangered and rescued two protocols with a combined TVL of over \$25 million from a 500+ days old bug.
+
+## **Background**
+<!-- truncate -->
+
+![](/img/blog-posts-img/a-zero-day-exploit-in-geb/moais.jpg)
```

markdownlint-cli2 (0.17.2)

Images should have alternate text (alt text)

(MD045, no-alt-text)